.docs/test_plan.md

```diff
@@ -0,0 +1,190 @@
+
+## functionalities
+
+- The submission of the tasks
+- The processing of the tasks
+- The retrieving of the results 
+
+# Prerequisite and exigences
+
+- Deploy ArmoniK with different partitions of bench (different number of pods in each partition)
+
+# The tools used to do the tests
+
+## Scripts bash
+
+- To start a warming up run then a batch of runs with the same parameters and store them in files
+
+## Python
+
+- Clean the files where we stored the data in readable json files.
+- Read the data and compute the median and the mean of each run batch.
+
+## Json 
+
+- We store the results of the performance tests in Json file which we will store in a database.
+
+# Tests environment
+
+- ArmoniK deployed on AWS via WSL2
+
+| Date | Infra version | Core version | API version  | Extension c# version | Instance type | Instance caracteristics | CPU frequency | nb vCPUs | RAM (GB) | Network bandwidth (Gbps) | EBS Bandwidth (Mbps) |
+| :---    | :---    | :---    | :---    | :---    | :---    | :---    | :---    | :---    | :---    | :---   |  :---   |
+| 03/29/2023 | 2.12.0 | 0.11.1 | 3.5.2 | 0.9.1 | c24.xlarge |  Intel Xeon de 2e (Cascade Lake) | 3.6 GHz - 3.9 GHz | 96 | 192 | 25 | 19000 |
+
+- [Config file](https://github.com/aneoconsulting/ArmoniK/blob/874-benchmarking/versions.tfvars.json) with all the versions of Armonik
+
+# Exploited ressources
+
+| Kubernetes | object storage type | storage queue type | storage table type | OS |
+| :---    | :---    | :---    | :---    | :---    |
+| AWS EKS "1.25" | AWS S3 | Amazon MQ, broker : ActiveMQ "5.16.4" | MongoDB "6.0.1" | Linux |
+
+# The estimated results
+
+- More pods -> faster treatement 
+
+# Tests to do
+
+## Throughput scalability tests
+
+- 1000 tasks on : 100 pods, 1000 pods, and 10000 pods
+- 5000 tasks on : 100 pods, 1000 pods, and 10000 pods
+- 10k tasks on : 100 pods, 1000 pods, and 10000 pods
+- 100k tasks on : 100 pods, 1000 pods, and 10000 pods
+- 1M tasks on : 100 pods, 1000 pods, and 10000 pods
+
+### with
+
+- tasks duration : 1 ms
+- payload size : 8B
+- result size : 8B
+
+#### StressTest
+
+| nb pods | nb cores per control-plane| limits nb cores per control-plane| nb cores per scheduler-agent| limits nb cores per scheduler-agent| nb cores per worker | limits nb cores per worker | nb tasks | task duration | input payload | result payload | Duration of submission (s) | Upload speed (s) | Throughtput for submission (tasks/s) | Duration of processing (s) | Throughput for processing (tasks/s) | Duration of retrieving results (s) | Throughtput for retrieving results (tasks/s) | Download speed (s) |
+| :---    | :---    | :---    | :---   |  :---   |  :----:  |    ---: |  :----:  |    ---: |    ---: |    ---: |    ---: |    ---: |    ---: |    ---: |    ---: |    ---: |    ---: |    ---: |
+| 100 | 200m  | 1000m | 1000m | 2000m | 50m | 1000m | 1K | 1 ms | 8B | 8B |  |  |  |  |  |  |  |  |
```

Comment:
Examples for resource requests and limits for each ArmoniK components:
For Resource requests: CPU: 200m / Memory: 500Mi
For Resource limits: CPU: 1000m / Memory: 2048Mi